### ISSUE TITLE
symbolic link cycles

### ISSUE THREAD

**Issue by brillout** [AurelienLourot/uncommitted#7]
Really nice tool, thanks 👍 

Got a small issue though. Upon `uncommitted -Luns .` I get
```
/home/romu/.config/nvim/.vim/.vim/.vim/.vim/.vim/.vim/.vim/.vim/.vim/.vim/.vim/.vim/.vim/.vim/.vim/.vim/.vim/.vim/.vim/.vim/.vim/.vim/.vim/.vim/.vim/.vim/.vim/.vim/.vim/.vim/.vim/.vim/.vim/.vim/.vim/.vim/.vim/.vim/.vim/plugged/fzf.vim - Git
?? doc/tags
```

I checked and `.vim/.vim` is a symlink loop:  `/home/romu/.vim/.vim -> /home/romu/.vim`

**Comment by AurelienLourot** [comment]
We use `os.walk()` without any max-depth-related option and we don't catch exceptions. I'll dig and try to find why `os.walk()` stops silently at an arbitrary depth without raising any error. Thanks for the issue.

**Comment by AurelienLourot** [comment]
`os.walk()`'s argument `onerror` doesn't help: no error is thrown.
I think I'll implement a solution similar to http://stackoverflow.com/a/36977656/1855917

**Comment by AurelienLourot** [comment]
This should now be fixed by 6b70854a0ef665c33e9fffea56e6bc4c8830e2f9 :)

**Comment by brillout** [comment]
cool thanks
